libs/datamodel/core/src/dml/model.rs

```diff
@@ -250,11 +250,32 @@ impl Model {
         result
     }
 
+    pub fn field_is_indexed(&self, field_name: &String) -> bool {
+        let field = self.find_field(field_name).unwrap();
+
+        if field.is_id() || field.is_unique() {
+            return true;
+        }
+
+        self.indices
```

Comment:
I found this one hard to understand because of the `||` on the new line. Assigning those two checks to variables like `is_start_of_index` and `is_start_of_multi_field_id` would probably ease that quite a bit.